src/Database/Expression/QueryExpression.php

```diff
@@ -699,38 +699,24 @@ protected function _addConditions(array $conditions, array $types): void
      * generating the placeholders and replacing the values by them, while storing
      * the value elsewhere for future binding.
      *
-     * @param string $field The value from which the actual field and operator will
+     * @param string $condition The value from which the actual field and operator will
      * be extracted.
      * @param mixed $value The value to be bound to a placeholder for the field
      * @return \Cake\Database\ExpressionInterface
      * @throws \InvalidArgumentException If operator is invalid or missing on NULL usage.
      */
-    protected function _parseCondition(string $field, mixed $value): ExpressionInterface|string
+    protected function _parseCondition(string $condition, mixed $value): ExpressionInterface|string
     {
-        $field = trim($field);
+        $expression = trim($condition);
         $operator = '=';
-        $expression = $field;
-
-        $spaces = substr_count($field, ' ');
-        // Handle operators with a space in them like `is not` and `not like`
-        if ($spaces > 1) {
-            $parts = explode(' ', $field);
-            if (preg_match('/(is not|not \w+)$/i', $field)) {
-                $last = array_pop($parts);
-                $second = array_pop($parts);
-                array_push($parts, strtolower("{$second} {$last}"));
-            }
-            $operator = array_pop($parts);
-            $expression = implode(' ', $parts);
-        } elseif ($spaces == 1) {
-            $parts = explode(' ', $field, 2);
-            [$expression, $operator] = $parts;
-            $operator = strtolower(trim($operator));
+        if (preg_match('/^(.*?)\s+([a-zA-Z=~!<>+\-*\\ ]+)$/', $condition, $matches)) {
```

Comment:
It would be good to test the performance of this vs string splitting. This is a very hot method in a non-trivial application.